app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
+            });
+        }
+        else if (this.mode_state === "PoolOJob"){
+            this.jobTemplate.job.properties.poolInfo = this.pickedPool.value;
+            const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this.jobTemplate, jobInputs]);
+            obs.subscribe({
+                next: (data) => this._redirectToJob(),
+                error: (err) => this.error = ServerError.fromPython(err),
+            });
+        }
+        else if (this.mode_state === "Pool"){
+            const obs = this.pythonRpcService.callWithAuth("create-ncj-pool", [this.poolTemplate, poolInputs]);
+            obs.subscribe({
+                next: (data) => this._redirectToPool(),
+                error: (err) => this.error = ServerError.fromPython(err),
+            });
+        }
 
-    private _getTemplates() {
-        this.templateService.getTemplates(this.applicationId, this.actionId).subscribe((templates) => {
-            this.jobTemplate = templates.job;
-            this.poolTemplate = templates.pool;
-            this._parseParameters();
-            this.createForms();
-        });
     }
 
-    private _buildJobTemplate(): any {
-        const template = { ...this.jobTemplate };
-        template.job.properties.poolInfo = this.pickedPool.value;
-        return template;
+    private _redirectToJob() {
+        this.router.navigate(["/jobs"]);
+    }
+    private _redirectToPool() {
+        this.router.navigate(["/pools"]);
     }
 
-    private _parseParameters() {
-        const parameters = this.jobTemplate.parameters;
-        const otherParameters: any[] = [];
-        for (let name of Object.keys(parameters)) {
-            const param = parameters[name];
-            if (name === ConventionNames.jobName) {
-                this.jobNameParam = new NcjParameterWrapper(name, param);
-            } else {
-                otherParameters.push(new NcjParameterWrapper(name, param));
+    private getContainerFromFileGroup(fileGroup: string) {
+        return fileGroup && `fgrp-${fileGroup}`;
+    }
+    private getType(param){
+        if (this.getParameters(this.jobTemplate).includes(param)){
+            if (this.jobTemplate.parameters[param].metadata.advancedType){
+                return this.jobTemplate.parameters[param].metadata.advancedType;
+            }
+            else if (this.jobTemplate.parameters[param].allowedValues){
+                return "dropdown";
+            }
+            return this.jobTemplate.parameters[param].type;
+        }
+        else if (this.getParameters(this.poolTemplate).includes(param)){
+            if (this.poolTemplate.parameters[param].metadata.advancedType){
+                return this.poolTemplate.parameters[param].metadata.advancedType;
             }
+            else if (this.poolTemplate.parameters[param].allowedValues){
+                return "dropdown";
+            }
+            return this.poolTemplate.parameters[param].type;
+        }
+        else {
```

Comment:
I may be reading this method incorrectly, but should we also have a check for 'int'? I don't know if that would be considered and 'advancedType' but I wouldn't think so.